### PR TITLE
E2E tests - Feature 1.1

### PR DESCRIPTION
**Pull request by andriizhegurov-okta**

### CODE REVIEW

File: src/A18NClient/A18nClient.cs [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+        }
+
+        public void SetDefaultProfileId(string profileId)
+        {
+            _defaultProfileId = profileId;
+        }
+
+        public void Dispose()
+        {
+            if (_client != null)
```
Comment (laura-rodriguez) [inline src/A18NClient/A18nClient.cs:40]: Let's implement the `Dispose(bool)` override, as suggested [here](https://docs.microsoft.com/en-us/dotnet/standard/garbage-collection/implementing-dispose).

File: src/Okta.Idx.Sdk.E2ETests/Drivers/IISWebServerDriver.cs [hunk 2]
```diff
@@ -0,0 +1,54 @@
+﻿using System;
+using System.Diagnostics;
+using System.IO;
+
+namespace Okta.Idx.Sdk.E2ETests.Drivers
+{
+    public class IISWebServerDriver: IWebServerDriver
+    {
+        private const int iisPort = 8080;
```
Comment (laura-rodriguez) [inline src/Okta.Idx.Sdk.E2ETests/Drivers/IISWebServerDriver.cs:9]: Should we read the port from the configuration file instead?

File: src/Okta.Idx.Sdk.E2ETests/Drivers/WebDriverDriver.cs [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    public class WebDriverDriver : IDisposable
+    {
+        private Lazy<IWebDriver> _webDriver = new(() => SetupWebDriver());
+
+        public IWebDriver WebDriver => _webDriver.Value;
+
+        private static IWebDriver SetupWebDriver()
+        {
+            var options = new ChromeOptions();
+//            options.AddArgument("--headless");
```
Comment (laura-rodriguez) [inline src/Okta.Idx.Sdk.E2ETests/Drivers/WebDriverDriver.cs:16]: Should we remove this line?

File: src/Okta.Idx.Sdk.E2ETests/Drivers/WebDriverDriver.cs [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+            options.AddArgument("--start-maximized");
+            options.AddArgument("--disable-notifications");
+            return new ChromeDriver(options);
+        }
+
+        public void Dispose()
```
Comment (laura-rodriguez) [inline src/Okta.Idx.Sdk.E2ETests/Drivers/WebDriverDriver.cs:22]: Same Dispose comment here.

File: src/Okta.Idx.Sdk.E2ETests/Features/11-BasicLoginWithPasswordFactor.feature [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+	Given a SPA, WEB APP or MOBILE Sign On Policy that defines Password as required
+	And the list of Authenticators contains Email and Password
+	And a User named "Mary" exists, and this user has already setup email and password factors
+
+  Scenario: 1.1.1 Mary logs in with a Password
+	Given Mary navigates to the Basic Login View
+	When she fills in her correct username
+	And she fills in her correct password
+	And she submits the Login form
+    Then she is redirected to the Root View
```
Comment (laura-rodriguez) [inline src/Okta.Idx.Sdk.E2ETests/Features/11-BasicLoginWithPasswordFactor.feature:14]: Nit: Fix indentation

File: src/Okta.Idx.Sdk.E2ETests/Helpers/TestUserHelper.cs [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+
+namespace Okta.Idx.Sdk.E2ETests.Helpers
+{
+    public class TestUserHelper : ITestUserHelper, IDisposable
+    {
+        private IA18nClient _a18nClient;
+        private ITestConfig _configuration;
+        private A18nProfile _a18nProfile;
+        private IOktaSdkHelper _oktaHelper;
+       // private IUser oktaUser;
```
Comment (laura-rodriguez) [inline src/Okta.Idx.Sdk.E2ETests/Helpers/TestUserHelper.cs:14]: SHould we remove this?

File: src/Okta.Idx.Sdk.E2ETests/Drivers/WebDriverDriver.cs (hunk 3 above)
Comment (andriizhegurov-okta) [reply src/Okta.Idx.Sdk.E2ETests/Drivers/WebDriverDriver.cs:16]: I think we may want to uncomment this line eventually, moving from local env to Travis CI.

File: src/Okta.Idx.Sdk.E2ETests/Steps/BasicLoginWithPasswordFactorSteps.cs [hunk 7]
```diff
@@ -0,0 +1,132 @@
...
+        }
+
+        [When(@"she fills in her incorrect username")]
+        public void WhenSheFillsInHerIncorrectUsername()
+        {
+            _loginPage.UserNameInput.SendKeys("wrongname@site.com");
+        }
+
+        [Then(@"she should see a message on the Login form ""(.*)""")]
+        public void ThenSheShouldSeeAMessageOnTheLoginForm(string p0)
```
Comment (laura-rodriguez) [inline src/Okta.Idx.Sdk.E2ETests/Steps/BasicLoginWithPasswordFactorSteps.cs:108]: Curious, is `p0` some sort of naming convention for these tests?

File: src/Okta.Idx.Sdk.E2ETests/settings.json [hunk 8]
```diff
@@ -0,0 +1,14 @@
+﻿{
+  "okta": {
+    "testing": {
+      "NormalUser": "mary@acme.com",
+      "DeactivatedUser": "mary-deactivated@acme.com",
+      "LockedUser": "mary-locked@acme.com",
+      "SuspendedUser": "mary-suspended@acme.com",
+      "UnassignedUser": "mary-unassigned@mailinator.com",
+      "UserPassword": "Abcd1234",
```
Comment (laura-rodriguez) [inline src/Okta.Idx.Sdk.E2ETests/settings.json:9]: I don't think we should expose these secrets. I suggest keeping them local and regenerate new ones.

File: src/Okta.Idx.Sdk.E2ETests/Steps/BasicLoginWithPasswordFactorSteps.cs (hunk 7 above)
Comment (andriizhegurov-okta) [reply src/Okta.Idx.Sdk.E2ETests/Steps/BasicLoginWithPasswordFactorSteps.cs:108]: It's autogenerated parameter name, I'm not using this parameter, I think I'll give those parameters better names

File: src/Okta.Idx.Sdk.E2ETests/settings.json (hunk 8 above)
Comment (andriizhegurov-okta) [reply src/Okta.Idx.Sdk.E2ETests/settings.json:9]: all the sensitive will be kept in secure Environment vars when we move to Travis CI, keeping them in the config file makes it easier to deploy tests on another local environment.

And actually those values are not used, they are leftovers from first version when I used pre-defined users, currently we are using A18N profiles. I will clean up those values.

File: src/Okta.Idx.Sdk.E2ETests/Drivers/WebDriverDriver.cs (hunk 3 above)
Comment (andriizhegurov-okta) [reply src/Okta.Idx.Sdk.E2ETests/Drivers/WebDriverDriver.cs:16]: removing the line, we know where to put it when we need it

**Review by laura-rodriguez — APPROVED** [review]

**Pull request merged**